hal/inc/hal_dynalib_core.h

```diff
@@ -79,6 +80,9 @@ DYNALIB_FN(32, hal_core, HAL_Set_Event_Callback, void(HAL_Event_Callback, void*)
 DYNALIB_FN(33, hal_core, HAL_Core_Enter_Stop_Mode_Ext, int(const uint16_t*, size_t, const InterruptMode*, size_t, long, void*))
 DYNALIB_FN(34, hal_core, HAL_Core_Execute_Standby_Mode_Ext, int(uint32_t, void*))
 
+DYNALIB_FN(35, hal_core, hal_sleep_validate_config, int(const hal_sleep_config_t*, void*))
```

Comment:
In that case, we need to allocate a `hal_sleep_config_t` structure from stack or heap to allow changing the dry-run flag, since the sleep configuration can be const from user aspect.

Reply:
That's a good point :+1:

Reply:
How about adding one more argument in `bool` to indicate dry-run or not?

Reply:
In that case it'd have to be instead probably an argument with bitmask flags, which might be confusing (with flags in `hal_sleep_config_t` and in function call). Let's keep a separate function.